app/vmalert/storage/storage.go

```diff
@@ -0,0 +1,11 @@
+package storage
+
+// Metrics the data returns from storage
+type Metrics struct{}
+
+// VMStorage represents vmstorage entity with ability to read and write metrics
+type VMStorage struct{}
+
+func (s *VMStorage) ReadMetrics(query string) ([]Metrics, error) {
```

Comment:
I'd suggest to rename this into `Query`.
The `Query` method should accept the `Rule` in arguments. This gives option for storage to decide how the query should be build. For example, if we'll have multiple storages, VM and Prometheus, their query text and params may be different.
Method should accept the Context for cancelation